Develop

JmKanmo opened the pull request:
07-23 ~ 07-30  작업 진행 
주요 작업 사항 

-  회원가입, 이메일 인증 코드 구현 (프론트엔드, 백엔드)
- sftp 프로토콜 사용 서비스 클래스, 메소드 작성 (추후 유저 프로필 사진, etc 정적 리소스 저장 용도)
- 비밀번호 재발급, 이메일 인증 기능 나머지 구현 (인증키, 인증기한 필드 추가)

프로젝트 기획 관련 참조 문서 
https://boiling-glasses-dec.notion.site/abfefa6fdd5b49e0818f0929b7d893ed

JmKanmo committed:
07-26 커밋
- 회원가입, 이메일 인증 코드 구현
-  프론트엔드 코드 수정

JmKanmo committed:
07-28 작업내용 커밋
- 사용자 회원가입, 이메일 인증 관련 코드 수정
- 로그인 시에 헤더 유저 프로필 업데이트 코드 작성
- sftp 프로토콜 사용 서비스 클래스, 메소드 작성

JmKanmo committed:
07-29 작업 내용
- 비밀번호 재발급, 이메일 인증 기능 나머지 구현 (인증키, 인증기한 필드 추가)
- 프론트단 소스코드 수정 및 재사용 fragment 개선
- 소스코드 전반적으로 리팩토링

JmKanmo committed:
디렉토리 구조 변경 (세분화)

JmKanmo committed:
springdoc-openapi 도입, API 페이지 작성

KimHyoJin commented on src/main/java/com/service/core/user/controller/UserController.java:
```diff
@@ -24,17 +29,16 @@
     }
 
-    @GetMapping("/user/complete")
+    @GetMapping("/user/signup_complete")
```
[Restful API guide](https://restfulapi.net/resource-naming/)에 따르면 _ 보다는 - 를 쓰는 것을 좀더 권장합니다-!

KimHyoJin commented on src/main/java/com/service/util/sftp/SFtpUtil.java:
```diff
@@ -0,0 +1,217 @@
+            if (fos != null) {
+                fos.flush();
+            }
+        } catch (Exception e) {
```
따로 처리해주는 부분이 없다면 굳이 catch를 안해줘도 괜찮을것 같습니다!

KimHyoJin commented on src/main/java/com/service/core/user/repository/UserRepository.java:
```diff
@@ -3,5 +3,10 @@
+import java.util.List;
+import java.util.Optional;
+
 public interface UserRepository extends JpaRepository<User, String> {
```
@Repository annotation이 빠진것 같아요

KimHyoJin commented on src/main/java/com/service/core/user/service/UserServiceImpl.java:
```diff
@@ -1,45 +1,171 @@
         grantedAuthorityList.add(new SimpleGrantedAuthority("ROLE_USER"));
 
-        return new org.springframework.security.core.userdetails.User(user.getUserId(), user.getPassword(), grantedAuthorityList);
+        return new org.springframework.security.core.userdetails.User(user.getEmail(), user.getPassword(), grantedAuthorityList);
```
`import com.service.core.user.domain.User`와 네이밍이 겹쳐서 풀네임을 적으신거같은데, naming을 다르게 가져가면 휴먼에러가 발새할 가능성이 좀더 적을 것 같습니다! 단순의견입니다~

KimHyoJin commented on src/main/java/com/service/util/JmUtil.java:
```diff
@@ -2,45 +2,83 @@
+import java.time.LocalDateTime;
+import java.util.UUID;
 
+public class JmUtil {
```
JmUtil은 혹시 어떤 뜻인가요?~

KimHyoJin commented on src/main/java/com/service/core/user/controller/UserController.java:
```diff
@@ -2,72 +2,223 @@
+
+            User user = User.from(signupForm);
+
+            if (userService.checkSameUser(user)) {
```
이런 로직부분은 controller가 아닌 service로 들어가는것이 좀더 좋을 것 같습니다.

KimHyoJin commented on src/main/java/com/service/config/SFtpConfig.java:
```diff
@@ -0,0 +1,27 @@
+
+@Component
+@Data
+public class SFtpConfig {
```
`ConfigurationProperties` annotation을 쓰면 좀더 간단히 mapping 가능할 것 같습니다

KimHyoJin commented on src/main/java/com/service/core/error/model/UserManageException.java:
```diff
@@ -0,0 +1,7 @@
+package com.service.core.error.model;
+
+public class UserManageException extends RuntimeException {
+    public UserManageException(String message) {
```
`UserManageException`에 `ConstUtil.UserAuthMessage`이 값이 담기는 것 같은데요, 강제하고 싶다면 (= `UserManageException`에 `ConstUtil.UserAuthMessage`에 정의한 에러만 담고싶다면) `ConstUtil.UserAuthMessage`를 enum으로 정의하여 사용해도 좋을 것 같습니다~!

KimHyoJin commented on src/main/java/com/service/core/user/controller/UserController.java:
```diff
@@ -2,72 +2,223 @@
-            model.addAttribute("error", String.format("회원 가입에 실패하였습니다. 발생 에러: %s", e.getMessage()));
+            userService.checkSameId(id);
+            return ResponseEntity.status(HttpStatus.OK).body("사용 가능한 id 입니다.");
+        } catch (UserManageException exception) {
```
`UserManageException`에 대한 catch를 `ControllerAdvice`에서 처리해도 좋을 것 같아요!

KimHyoJin commented on src/main/resources/application.yml:
```diff
@@ -55,11 +78,23 @@ logging:
 #    org.springframework.web: DEBUG
 #
-
 mybatis:
```
myBatis와 JPA가 혼용된 이유가 궁금합니다!

JmKanmo replied:
QueryDSL을 써보려 햇는데 환경설정 하기가 힘들고 그래서 포기햇엇습니다. 아무튼 사용하는데 진입장벽이  
mybatis가 좀 더 사용하기 쉽고 native query 작성할수잇는 장점이 잇고 해서 끌려서 적용햇습니다 
추후 진행할 다른 프로젝트에선 QueryDSL도 사용해볼 예정입니다

JmKanmo replied:
이참에 지난번에 포기했던 QueryDSL 적용을 한번 더 시도 해봐야겟네요.

KimHyoJin commented on src/main/java/com/service/core/blog/service/BlogServiceImpl.java:
```diff
@@ -8,10 +8,11 @@
 @Service
 @RequiredArgsConstructor
-@Transactional
+@Transactional(readOnly = true)
```
이부분의 `@Transactional`은 어떤이유로 설정하신건가용?

KimHyoJin reviewed (approved):
와 많은 코드 작성하셨네요~!! 코드양이 생각보다 많아서 전체적으로 한번 리뷰해보았습니다! 
수고 많으셨습니다 👍🏻

JmKanmo committed:
- 코드리뷰 반영

JmKanmo committed:
서비스단 junit 테스트 코드 작성

JmKanmo committed:
- spring redis 추가
- user service 코드 수정 (UserInfoService, UserAuthService) 분리
- UserInfoService <-  Jpa 비즈니스로직,  UserAuthService - Redis 비지니스 로직

JmKanmo committed:
- user service 테스트 코드 작성
- UserAuthService 코드 수정

KimHyoJin commented:
준모님! 하나의 PR에 여러 기능을 넣어도 좋지만 좀더 잘게 쪼개면 리뷰하기가 편할것 같습니다 🥰 한번 리뷰 받고 어느정도 마무리가 되면 머지 후 새로 브랜치를 따서 작업하시는걸 추천드려요! ( 내용이 너무 많아지면 리뷰가 어려워서요ㅠ)

JmKanmo committed:
08.03 ~ 08.09
- 설정 페이지 구성 (프론트엔드 + 백엔드)
- spring redis 추가 및 세션,캐싱 기능 적용 (이메일 인증, 비밀번호 재설정 id,key로 구성 및 캐싱)
- 소스코드, API 리팩토링
- 아래 기능 추가
   - 프로필 이미지 업로드 및 삭제
   - 회원탈퇴
   - 비밀번호 변경
   - 소개글 추가 및 사용자 기본 정보 수정
